client/src/modules/SortList/SortList.tsx

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useRef, useState } from 'react';
+import { categories } from '../../assets/sortByList';
+import { useAppDispatch, useAppSelector } from '../../hooks/redux-hooks';
+import { selectAll } from '../../redux/selectors/todolistSelectors';
+import { setSortBy } from '../../redux/slices/todoListSlice';
+import { SortTypes } from '../../types';
+import styles from './SortList.module.scss';
+
+const SortList = () => {
+  const { wrapper, container, item, active, icon } = styles;
+
+  const ref = useRef<HTMLElement>(null);
+  const dispatch = useAppDispatch();
+  const { title } = useAppSelector(selectAll).sortBy;
+  const [showSort, setShowSort] = useState<boolean>(false);
+
+  useEffect(() => {
+    const closeSortList = (e: MouseEvent) => {
+      const path = e.composedPath();
+      if (ref.current) {
+        if (!path.includes(ref.current)) {
+          setShowSort(false);
+        }
+      }
+    };
```

Comment:
You can take function out of useEffect, you can describe it in upper scope. Keep useEffect clean also